crates/iota/src/client_ptb/ptb.rs

```diff
@@ -54,19 +54,33 @@ pub struct Summary {
 impl PTB {
     /// Parses and executes the PTB with the sender as the current active
     /// address
-    pub async fn execute(self, context: &mut WalletContext) -> Result<(), Error> {
+    pub async fn execute(self, context: &mut WalletContext) -> Result<String, Error> {
+        let res = self.execute_to_styled_str(context).await?;
+        println!("{res}");
+        Ok(res.to_string())
+    }
+
+    /// Parses and executes the PTB with the sender as the current active
+    /// address
+    pub async fn execute_to_styled_str(
```

Comment:
This should maybe be private?

Reply:
This whole crate is not really mean to be used by others so I would say it doesn't matter, but I can change to private if you prefer that

Reply:
I would personally still make it private but no strong feeling